Checkout cookbook (#1296)
saleor/saleor-docs: docs/developer/checkout/cookbook.mdx

```
</TabItem>
</Tabs>

## Partial/Split payments

> Common use cases of splitting payments on a single order are:

Reviewer: Common example would be paying part with gift card, and rest with credit card

- Charging only for part of the order and another part after the fulfillment.
- Orders are split into fulfillments paid separately to each vendor.
- Authorize part of the basket as a pre-order payment (without charging) and change it immediately before fulfillment.

> #### Possible Approach

Reviewer: ```suggestion
### Possible Approach
```

:::

## Product personalization

Common use cases of product personalization are:

- Customized products (e.g., engraved jewelry, custom t-shirts)
- Packaging preferences
- Product configuration such as PC components, furniture, cars, etc.
- Delivery preferences for each item

The personalization can be achieved by adding additional [`metadfields`](/api-reference/checkout/objects/checkout-line.mdx#checkoutlinemetafieldstring--) fields to the [`CheckoutLine`](/api-reference/checkout/objects/checkout-line.mdx) object.

With the following mutations:
- [`checkoutLinesAdd`](/api-reference/checkout/mutations/checkout-lines-add.mdx)
- [`checkoutLinesUpdate`](/api-reference/checkout/mutations/checkout-lines-update.mdx)

The [`metdafields`](/api-reference/orders/objects/order-line.mdx#orderlinemetafieldstring--) will be copied to the [`OrderLine`](/api-reference/orders/objects/order-line.mdx) after checkout completion.

Additional steps might be required to process such fields in the fulfillment process, such as:

- Pass the fields to the ERP system
- Listen to [`webhook`](/developer/extending/webhooks/overview.mdx) events such as `ORDER_CREATED` to process the fields

> - [Custom pricing](#custom-pricing)

Reviewer: Could be worth noting that metadata can be edited without permissions, which means those should be validated on server side if used for critical things like shipping preferences or variant selection.

Author: Or write from the server to `privateMetadata` 👍

## Custom product pricing

To set prices on checkout lines dynamically you can use the [`checkoutLinesUpdate`](/api-reference/checkout/inputs/checkout-line-update-input.mdx#checkoutlineupdateinputpricepositivedecimal--) mutation.
See example [repository](https://github.com/saleor/saleor-app-checkout-prices) for creating custom pricing middleware.

## Using phone number instead as identity

> In some cases, you might want to use a phone number instead of an email address for customer identification. While Saleor always requires email to create orders, you can email field as a phone number or other semantic meaning. For example: `123456789@phone`.

Reviewer: It should be mentioned to always use domain controlled by the developers, to avoid PII leak if those addresses will be used for mass mailing.

> In some cases, you might want to use a phone number instead of an email address for customer identification. While Saleor always requires email to create orders, you can email field as a phone number or other semantic meaning. For example: `123456789@phone`.

Reviewer: ```suggestion
In some cases, you might want to use a phone number instead of an email address for customer identification. While Saleor always requires email to create orders, you can email field as a phone number or other semantic meaning. For example: `123456789@noreply.yourcompany.com`.
```

## Order approvals and quotes

Example order approval flow:

> - Customer places an order without payment
> - Admin reviews the order and provides a final price in the form of a discount
> - Customer pays the final price
> - The order is shipped

Reviewer: ```suggestion
1. Customer places an order without payment
2. Admin reviews the order and provides a final price in the form of a discount
3. Customer pays the final price
4. The order is shipped
```

[Channel setting](/developer/channels/configuration.mdx#allow-unpaid-orders) should enable [`allowUnpaidOrders`](/api-reference/orders/inputs/order-settings-input.mdx#ordersettingsinputallowunpaidordersboolean--) to create orders without payments, (settings can be set via [API](/api-reference/channels/mutations/channel-update.mdx) or _Dashboard -> Order -> Order settings cogwheel_).

To require manual approval of orders in the dashboard, set the [`automaticallyConfirmAllNewOrders`](/api-reference/orders/inputs/order-settings-input.mdx#ordersettingsinputautomaticallyconfirmallnewordersboolean--) to `false`
(can be set via the dashboard _Configuration -> Channel_).

To arrange communication between flow between admin see Order related [`webhooks`](/api-reference/webhooks/enums/webhook-sample-event-type-enum.mdx).

## Subscriptions

Common use cases of subscriptions are:

- Create a subscription for a product that gets fulfilled periodically.
- Create membership subscriptions that do not require fulfillment.

A subscription service can be implemented as a standalone service that communicates with Saleor to write orders, update payments, and fulfill orders.

> If your admins need to manage subscriptions, you can use Saleor's dashboard to create a dedicated UI in the Saleor dashboard that will talk to your subscription service.

Reviewer: ```suggestion
If your admins need to manage subscriptions, you can use custom app to create a dedicated UI in the Saleor dashboard that will talk to your subscription service.
```

Might be worth linking SDK page explaining how to do this
